src/driver/amdxdna/aie2_pci.c

```diff
@@ -1133,7 +1133,9 @@ const struct amdxdna_dev_ops aie2_ops = {
 	.get_aie_info		= aie2_get_info,
 	.set_aie_state		= aie2_set_state,
 	.hwctx_init		= aie2_hwctx_init,
-	.hwctx_fini		= aie2_hwctx_fini,
+	.hwctx_flush_async	= aie2_hwctx_flush_async,
+	.hwctx_free		= aie2_hwctx_free,
+	.hwctx_fini		= NULL,
```

Comment:
Why the hwctx_fini is needed here?  Maybe just add a _free callback?

Reply:
I will remove hwctx_fini, when our internal devices also moved to hwctx_flush_async and hwctx_free.